src/commands/import.js

```diff
@@ -0,0 +1,187 @@
+const { flags } = require('@oclif/command')
+const FaunaCommand = require('../lib/fauna-command.js')
+const FaunaWriteStream = require('../lib/fauna-write-stream')
+const faunadb = require('faunadb')
+const path = require('path')
+const q = faunadb.query
+
+class ImportCommand extends FaunaCommand {
+  async run() {
+    const { db } = this.flags
+    const { source } = this.args
+    const { client } = await (db
+      ? this.ensureDbScopeClient(db)
+      : this.getClient())
+    this.client = client
+
+    this.log(`Database${db ? `'${db}'` : ''} connection established`)
+
+    const isDir = fs.lstatSync(source).isDirectory()
+
+    return (isDir ? this.importDir() : this.importFile(source)).catch((error) =>
+      this.handleError(error)
+    )
+  }
+
+  async importDir() {}
+
+  async importFile(source) {
+    let { collection } = this.flags
+    const parsedFileName = this.parseFileName(source)
+    if (!collection) {
+      collection = parsedFileName.name
+    }
+    await this.dataImport({ source, collection })
+  }
+
+  parseFileName(source) {
+    const { name, ext } = path.parse(path.basename(source))
+
+    // TODO: support json
+    if (ext !== '.csv') {
+      throw new Error(`File (${source}) must include the '.csv' extension.`)
+    }
+    return { name, ext }
+  }
+
+  async dataImport({ source, collection }) {
+    const { ref, isCollectionExists } = await this.upsertCollection({
+      collection,
+    })
+
+    if (isCollectionExists) {
+      await this.clearCollectionDocuments(ref)
+    }
+
+    const faunaWriteStream = new FaunaWriteStream({
+      source,
+      log: this.log,
+      collectionRef: ref,
+      client: this.client,
+    })
+
+    await new Promise((resolve, reject) => {
+      fs.createReadStream(source)
+        .pipe(csv.createStream())
+        .pipe(faunaWriteStream)
+        .on('error', reject)
+        .on('end', resolve)
+    })
+  }
+
+  handleError(error) {
+    if (error instanceof faunadb.errors.FaunaHTTPError) {
+      this.error('Error:', error.faunaError.message)
+      return this.error(
+        util.inspect(error, {
+          depth: null,
+          compact: false,
+        })
+      )
+    }
+    if (error instanceof Error) {
+      return this.error(error.message)
+    }
+
+    if (Array.isArray(error)) {
+      this.error(`\r\n\t${error.join('\r\n\t')}`)
+    }
+
+    this.error(error)
+  }
+
+  upsertCollection({ collection }) {
+    return this.client
+      .query(
+        q.Let(
+          {
+            ref: q.Collection(collection),
+            isCollectionExists: q.Exists(q.Var('ref')),
+            collection: q.If(
+              q.Var('isCollectionExists'),
+              '',
+              q.Let(
+                {
+                  collection: q.CreateCollection({ name: collection }),
+                },
+                q.CreateIndex({
+                  source: q.Select(['ref'], q.Var('collection')),
+                  name: `all_${collection}`,
+                })
```

Comment:
What's the reason for this index? It sounds like you could  just use the `Documents(..)` instead.